### ISSUE TITLE
Inline volume support for nfs protocol to azure storage account blob

### ISSUE THREAD

**Issue by AnandKatta** [kubernetes-sigs/blob-csi-driver#1022]

**Comment by AnandKatta** [comment]
Is the inline volume supported for nfs protocol.

**Comment by andyzhangx** [comment]
yes,  set `protocol: nfs` in `volumeAttributes`
```
  volumes:
    - name: persistent-storage
      csi:
        driver: blob.csi.azure.com
        volumeAttributes:
          containerName: EXISTING_CONTAINER_NAME  # required
          protocol: nfs
```

**Comment by AnandKatta** [comment]
How to add the secret reference to volume attributes, it is not pulling the credentials. Please share some reference link.

Just for confirmation, here it says it was not supported(https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/deploy/example/e2e_usage.md#:~:text=only%20available%20from%20v1.2.0%20for%20blobfuse%20protocol%20(NFS%20protocol%20is%20not%20supported)
Thanks for quick reference.

**Comment by andyzhangx** [comment]
> How to add the secret reference to volume attributes, it is not pulling the credentials. Please share some reference link.
> 
> Just for confirmation, here it says it was not supported(https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/deploy/example/e2e_usage.md#:~:text=only%20available%20from%20v1.2.0%20for%20blobfuse%20protocol%20(NFS%20protocol%20is%20not%20supported) Thanks for quick reference.

@AnandKatta nfs protocol is also supported with inline volume, and nfs does not require credentials, it requires network config on the nfs storage account, follow guide here: https://github.com/kubernetes-sigs/blob-csi-driver/tree/master/deploy/example/nfs#prerequisite, I will update the doc later.

**Comment by AnandKatta** [comment]
Thanks !!..

@andyzhangx .The link  is not useful for inline volume mounting. I am not pretty sure, where to pass storage account name and resource group name without storage class.

Thanks again

**Comment by andyzhangx** [comment]
@AnandKatta  try this config:
```
  volumes:
    - name: persistent-storage
      csi:
        driver: blob.csi.azure.com
        volumeAttributes:
          storageAccount: EXISTING_ACCOUNT_NAME
          containerName: EXISTING_CONTAINER_NAME  # required
          protocol: nfs
```

**Comment by AnandKatta** [comment]
It works with "storageaccount" along with "storageAccount"

**Comment by calohmn** [comment]
> > Is the inline volume supported for nfs protocol.
> 
> yes, set `protocol: nfs` in `volumeAttributes`

@andyzhangx It would be good to update
https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/deploy/example/e2e_usage.md#option3-inline-volume
then, there it still says:
```
Option#3: Inline volume

    only available from v1.2.0 for blobfuse protocol (NFS protocol is not supported)
```
